nf_core/modules/module_test.py

```diff
@@ -71,17 +76,27 @@ def run(self):
     def _check_inputs(self):
         """Do more complex checks about supplied flags."""
 
+        # Get installed modules
+        self.all_local_modules, self.all_nfcore_modules = nf_core.modules.module_utils.get_installed_modules(
+            ".", "modules"
+        )
+
         # Get the tool name if not specified
         if self.module_name is None:
             if self.no_prompts:
                 raise UserWarning(
                     "Tool name not provided and prompts deactivated. Please provide the tool name as TOOL/SUBTOOL or TOOL."
                 )
-            modules_repo = ModulesRepo()
-            modules_repo.get_modules_file_tree()
+            all_installed_modules = [m.module_name for m in self.all_nfcore_modules]
```

Comment:
Maybe it should be merged here? I'll make a proposal:
```suggestion
            all_installed_modules = (
                *(m.module_name for m in self.all_nfcore_modules),
                *(m.module_name for m in self. all_local_modules),
            )
```

Reply:
The problem with local modules is that they are retrieved differently by get_installed_modules(), outputting the full path to the main.nf file, as the folder structure is different, mainly because they are not expected to have tests.
Someone can add tests, but I think that this is a particular case, so I don't want to change the full function for that, that's why I implemented this code using only nf-core modules.

Currently, the command needs to find the proper folder structure, being modules/<toolname> and tests/<toolname>, so you must execute that from inside modules/nf-core or modules/local. As the user needs to create all this folder structure to add tests, I don't think it's a big handicap to have to change directories to run them.

If you have any other suggestion on how to approach this, let's discuss it! I don't think it will be common to have tests for local modules, or even nf-core modules inside a pipeline. Maybe adding a new function to retrieve modules when the command is run from a pipeline directory, and checking that the folder structure is the correct one...

Reply:
Also, I am now thinking that this will change the import of modules inside the pipeline code, which is not a big deal, only relative paths will change, but I will need to check if this affects something else, I am not sure right now.

Maybe a good option would be to add a different get_installed_modules() function for pipeline modules and change directories automatically before running pytest.

Reply:
I noticed also that the `get_installed_modules` only fetches modules installed from the `nf-core/modules` repository. I think it make sense to configure this command in the same way as the other modules commands, so it can be run with the `--git-remote` flag and so on. Then we should add a method to the `ModulesRepo` that checks which modules are installed from a repo in the pipeline, and use them for prompting.